src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -138,7 +138,7 @@ public function denormalize(mixed $data, string $type, ?string $format = null, a
 
     public function supportsDenormalization(mixed $data, string $type, ?string $format = null, array $context = []): bool
     {
-        return isset(self::SUPPORTED_TYPES[$type]);
+        return $context[self::SUPPORTED_TYPES_KEY][$type] ?? $this->defaultContext[self::SUPPORTED_TYPES_KEY][$type] ?? false;
```

Comment:
relying on the context in `supportsDenormalization` would require marking the types as non-cacheable in `getSupportedTypes` as the result is not cacheable based on the type and format alone.

Reply:
@stof Indeed the types would not be cache-able. But I'm not sure caching would be a good idea in the first place, because of potential memory leak concerns.